### PR TITLE
adding full search component

### PR DESCRIPTION
**Pull request by Fadiabb**

### CODE REVIEW

File: client/src/styles.scss [hunk 1]
```diff
@@ -4,10 +4,12 @@
 /** Import brand theme and (new) component themes */
 @import './assets/styles/openslides-theme';
 @import './app/site/site.component.scss-theme';
+@import './app/site/motions/components/motion-detail/motion-detail.component.scss-theme.scss';
```
[the remark this answers is not in the document]
Comment (emanuelschuetze) [reply client/src/styles.scss]: Works with your branch `themes`. But please check if the theme.scss should be contains in this full text search pull request.

File: client/src/app/site/full-search/full-search.component.html [hunk 2]
```diff
@@ -0,0 +1,85 @@
+<os-head-bar [nav]="false" [backButton]=true>
+
+    <!-- Title -->
+    <div class="title-slot">
+        <h2 translate>Search results</h2>
+    </div>
+    <div class="menu-slot">
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: Add Comment: (i.e)

File: client/src/app/site/full-search/full-search.component.html [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        <button type="button" mat-icon-button [matMenuTriggerFor]="menu">
+            <mat-icon>more_vert</mat-icon>
+        </button>
+    </div>
+</os-head-bar>
+
+
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: a lot of empty lines.

File: client/src/app/site/full-search/full-search.component.html [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        <mat-checkbox class="check-box" [(ngModel)]="assigmentCheck"> Election </mat-checkbox>
+    </button>
+</mat-menu>
+
+<mat-list *ngIf="user.length && userCheck">
+    <h2 mat-subheader> Users </h2>
+    <mat-nav-list>
+        <a mat-list-item *ngFor="let myUser of user">
+            <h5 mat-line routerLink="{{myUser.getDetailStateURL()}}">
+                    <mat-card> {{ myUser }}</mat-card>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: one whitesprace too much:
`<mat-card> {{`

File: client/src/app/site/full-search/full-search.component.html [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        <mat-divider></mat-divider>
+    </mat-nav-list>
+</mat-list>
+
+<mat-list *ngIf="motion.length && motionCheck">
+    <h2 mat-subheader>motions</h2>
+    <mat-nav-list>
+        <a mat-list-item *ngFor="let myMotion of motion">
+            <h5 mat-line routerLink="{{myMotion.getDetailStateURL()}}">
+               <mat-card> {{ myMotion }}</mat-card>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: as abobe, one whitesprace too much

File: client/src/app/site/full-search/full-search.component.html [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    <button mat-menu-item>
+        <mat-checkbox class="check-box" [(ngModel)]="agendaCheck"> Agneda </mat-checkbox>
+    </button>
+    <button mat-menu-item>
+        <mat-checkbox class="check-box" [(ngModel)]="assigmentCheck"> Election </mat-checkbox>
+    </button>
+</mat-menu>
+
+<mat-list *ngIf="user.length && userCheck">
+    <h2 mat-subheader> Users </h2>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: add translate annotation and remove whitespace

File: client/src/app/site/full-search/full-search.component.html [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            <h5 mat-line routerLink="{{myUser.getDetailStateURL()}}">
+                    <mat-card> {{ myUser }}</mat-card>
+            </h5>
+        </a>
+        <mat-divider></mat-divider>
+    </mat-nav-list>
+</mat-list>
+
+<mat-list *ngIf="motion.length && motionCheck">
+    <h2 mat-subheader>motions</h2>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: Add translate annotation and write "motions" with capital M

File: client/src/app/site/full-search/full-search.component.html [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+            <h5 mat-line routerLink="{{myMotion.getDetailStateURL()}}">
+               <mat-card> {{ myMotion }}</mat-card>
+            </h5>
+        </a>
+        <mat-divider></mat-divider>
+    </mat-nav-list>
+</mat-list>
+
+<mat-list *ngIf="agenda.length && agendaCheck">
+    <h2 mat-subheader>Agenda</h2>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: Add translate annotation

File: client/src/app/site/full-search/full-search.component.html [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+    <mat-nav-list>
+        <a mat-list-item *ngFor="let myAgenda of agenda">
+            <h5 mat-line routerLink="/agenda">
+                <mat-card> {{ myAgenda }}</mat-card>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: As above: one whitespace too much

File: client/src/app/site/full-search/full-search.component.html [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+            </h5>
+        </a>
+        <mat-divider></mat-divider>
+    </mat-nav-list>
+</mat-list>
+
+<mat-list *ngIf="assigment.length">
+    <h2 mat-subheader>Assigment</h2>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: Add translate annotation

File: client/src/app/site/full-search/full-search.component.html [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    <mat-nav-list>
+        <a mat-list-item *ngFor="let myAssigment of assigment">
+            <h5>
+                    <mat-card> {{ myAssigment }}</mat-card>
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.html]: indentation seems too large (or github plays tricks on me).
Also: Whitespace

File: client/src/app/site/full-search/full-search.component.scss [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+    flex-direction: column;
+}
+
+.search-feld > * {
+    width: 80%;
+}
+
+.check-box {
+    margin: 10px;
+}
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.scss]: add empty line

File: client/src/app/site/full-search/full-search.component.spec.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { FullSearchComponent } from './full-search.component';
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.spec.ts]: need to imort the e2e imports module. check other `*.spec.ts` files to get the idea

File: client/src/app/site/full-search/full-search.component.ts [hunk 14]
```diff
@@ -0,0 +1,77 @@
+import { Component, OnInit } from '@angular/core';
+import { DataStoreService } from 'app/core/services/data-store.service';
+import { User } from 'app/shared/models/users/user';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Motion } from 'app/shared/models/motions/motion';
+import { Item } from 'app/shared/models/agenda/item';
+import { Assignment } from 'app/shared/models/assignments/assignment';
+
+@Component({
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.ts]: classes, functions and constructors need to be commented using js docstring.

File: client/src/app/site/full-search/full-search.component.ts [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+    public motionCheckbox: true;
+    public userCheck = true;
+    public motionCheck = true;
+    public agendaCheck = true;
+    public assigmentCheck = true;
+    public term: string;
+
+    public constructor(private DS: DataStoreService, private route: ActivatedRoute, private router: Router) {}
+
+    public search(): void {
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.ts]: Comment

File: client/src/app/site/full-search/full-search.component.ts [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+            return this.searchInModel<Item>(item);
+        });
+
+        // search in assigment
+        this.assigment = this.DS.filter<Assignment>(Assignment, assigment => {
+            return this.searchInModel<Assignment>(assigment);
+        });
+    }
+
+    private searchInModel<T>(model: T, excludeSearch?: string): boolean {
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.ts]: comment

File: client/src/app/site/full-search/full-search.component.ts [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+        let found = false;
+        const keys = Object.keys(model);
+        keys.forEach(key => {
+            if (model[key] && typeof model[key] === 'string' && !found && key !== excludeSearch) {
+                found = model[key].includes(this.searchQuery);
+            }
+        });
+        return found;
+    }
+    public quickSearch(term: string): void {
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.ts]: comment + missing empty line

File: client/src/app/site/full-search/full-search.component.ts [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+        this.searchQuery = term;
+        this.search();
+        console.log('quick search for ', this.searchQuery);
+    }
+
+    public ngOnInit(): void {
```
Comment (tsiegleauq) [inline client/src/app/site/full-search/full-search.component.ts]: comment

File: client/src/app/site/site.component.html [hunk 19]
```diff
@@ -57,8 +57,17 @@
             <a [@navItemAnim] *osPerms="'core.can_see_projector'" mat-list-item routerLink='/projector'
                 routerLinkActive='active' (click)='toggleSideNav()'>
                 <mat-icon>videocam</mat-icon>
-                <span translate>Projector</span>
+                <span translate>Projektor</span>
```
Comment (tsiegleauq) [inline client/src/app/site/site.component.html]: Projector was right ;)

File: client/src/app/site/site.component.ts [hunk 20]
```diff
@@ -57,6 +59,7 @@ export class SiteComponent extends BaseComponent implements OnInit {
     public constructor(
         private authService: AuthService,
         private router: Router,
+        private DS: DataStoreService,
```
Comment (tsiegleauq) [inline client/src/app/site/site.component.ts]: can be removed again

File: client/src/app/site/site.component.ts [hunk 21]
```diff
@@ -10,6 +10,8 @@ import { pageTransition, navItemAnim } from 'app/shared/animations';
 import { MatDialog, MatSidenav } from '@angular/material';
 import { ViewportService } from '../core/services/viewport.service';
 import { MainMenuService } from '../core/services/main-menu.service';
+import { User } from 'app/shared/models/users/user';
```
Comment (tsiegleauq) [inline client/src/app/site/site.component.ts]: can be removed again

File: client/src/app/site/site.component.ts [hunk 22]
```diff
@@ -10,6 +10,8 @@ import { pageTransition, navItemAnim } from 'app/shared/animations';
...
+import { DataStoreService } from 'app/core/services/data-store.service';
```
Comment (tsiegleauq) [inline client/src/app/site/site.component.ts]: can be removed again

**Comment by FinnStutzenstein** [comment]
Please review. The design might be improved later.

**Pull request merged**